src/profile/sigmapr.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    refitσ!(m::LinearMixedModel{T}, σ::T, tc::TableColumns{T}, obj::T, neg::Bool)
+
+Refit the model `m` with the given value of `σ` and return a NamedTuple of information about the fit.
+
+`obj` and `neg` allow for conversion of the objective to the `ζ` scale and `tc` is used to return a NamedTuple
+"""
+function refitσ!(
+    m::LinearMixedModel{T}, σ, tc::TableColumns{T}, obj::T, neg::Bool
+) where {T}
+    m.optsum.sigma = σ
+    refit!(m; progress=false)
+    return mkrow!(tc, m, (neg ? -one(T) : one(T)) * sqrt(m.objective - obj))
+end
+
+"""
+    _facsz(m, σ, objective)
+
+Return a factor such that refitting `m` with `σ` at its current value times this factor gives `ζ ≈ 0.5`
+"""
+function _facsz(m::LinearMixedModel{T}, σ::T, obj::T) where {T}
+    i64 = inv(64)
+    m.optsum.sigma = σ * exp(i64)
+    return exp(i64 / (2 * sqrt(refit!(m; progress=false).objective - obj)))
+end
+
+"""
+    profileσ(m::LinearMixedModel, tc::TableColumns; threshold=4)
+
+Return a Table of the profile of `σ` for model `m`.  The profile extends to where the magnitude of ζ exceeds `threshold`.
+"""
+function profileσ(m::LinearMixedModel{T}, tc::TableColumns{T}; threshold=4) where {T}
+    (; σ, optsum) = m
+    isnothing(optsum.sigma) ||
+        throw(ArgumentError("Can't profile σ, which is fixed at $(m.optsum.sigma)"))
+    θ = copy(optsum.final)
+    θinitial = copy(optsum.initial)
+    _copy_away_from_lowerbd!(optsum.initial, optsum.final, optsum.lowerbd)
```

Comment:
I wonder if we should reduce the size of the initial step? The default initial step probably takes us quite far from the optimum, assuming that the optimum in the profiling steps is close-ish to the MLE.